src/Editor/Editor.cpp

```diff
@@ -225,6 +227,94 @@ void Editor::Show(float deltaTime) {
     if (currentEntity != nullptr) {
         glm::mat4 currentEntityMatrix = currentEntity->GetLocalMatrix();
 
+
+        currentEntityMatrix = currentEntity->GetLocalMatrix();
+
+        // Projection matrix.
+        glm::mat4 projectionMatrix = cameraEntity->GetComponent<Component::Lens>()->GetProjection(glm::vec2(ImGui::GetWindowSize().x, ImGui::GetWindowSize().y));
+
+        // View matrix.
+        glm::mat4 viewMatrix = glm::inverse(cameraEntity->GetModelMatrix());
+
+        // Load Vertices and Indices for the ray-triangle intersection test when the "paint mode" button is pressed.
+        if (currentEntity->loadPaintModeClicked && currentEntity->vertsLoaded) {
+
+            Geometry::Model * model = dynamic_cast<Geometry::Model*>(currentEntity->GetComponent<Component::Mesh>()->geometry);
```

Comment:
Asterisk goes to the left. Also, `model` must be checked for validity because it's the result of a `dynamic_cast`.